[Drilldows] Url Drilldown basic template helpers

Dosant opened the pull request:
## Summary

close https://github.com/elastic/kibana/issues/77758


### New helpers 

Provides a basic list of string manipulation helpers for URL Drilldown templating: 

| Helper  | Description | 
| ------------- | ------------- | 
| `formatNumber`  | A wrapper around [elastic/numeral-js](https://github.com/elastic/numeral-js). Allows to format number, currencies, bytes and etc...  `{{formatNumber value "0.0"}}` | 
| `lowercase`  | transform a string to lowercase string |
| `uppercase`  | transform a string to uppercase string |
| `trim`  |  |
| `trimLeft`  |  |
| `trimRight`  |   | 
| `left`  | `{{left "12345" 3}} -> "123"`  | 
| `right`  | `{{right "12345" 3}} -> "345"`  | 
| `mid`  | `{{mid "12345" 1 3}} -> "234"`  | 
| `concat`  | concatenates list of values into a string: `{{concat value1 ";" value2 }}`  | 
| `replace`  |  Can be used to replace a substring with another substring. Replaces all occurrences. `{{replace "Label:Feature:Something" "Label:" ""}} -> "Feature:Something"` | 
| `split`  |  Splits a string using a splitter character `{{split "Label:Feature:Something" ":"}} -> ["Label", "Feature", "Something"]` | 



### Checklist

Delete any items that are not applicable to this PR.

- [x] [Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html) was added for features that require explanation or tutorials - *Will add use facing documentation when we agree with the list*
- [x] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios

### For maintainers

- [x] This was checked for breaking API changes and was [labeled appropriately](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/81585/)
* Commit: 10c50172d90ad5daa6f7adbba24087dfd8501228

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### page load bundle size
| id | [before](https://github.com/elastic/kibana/commit/10c50172d90ad5daa6f7adbba24087dfd8501228) | [after](https://github.com/elastic/kibana/commit/2955584fc9ea77bfcabd1d0cff92ffe2ac6cea5b) | diff |
| --- | --- | --- | --- |
| `uiActionsEnhanced` | 327.7KB | 330.2KB | +2.6KB |

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/81585/","number":"81585","commit":"10c50172d90ad5daa6f7adbba24087dfd8501228"}]}
PIPELINE-->

elasticmachine commented:
Pinging @elastic/kibana-app-arch (Team:AppArch)

elastic-jb commented:
This is a great set to handle most simple string manipulations. I think we may want to still consider a way for a developer to register a custom helper if it's useful for them, but that is a separate goal.

Dosant commented:
> a way for a developer to register a custom helper if it's useful for them, but that is a separate goal.

@elastic-jb 👍 this is tracked here: https://github.com/elastic/kibana/issues/78404

elastic-jb commented:
I think we need the change for event.row (79157) to make use of the numeric formatting. I can't create a drilldown on a table that only has an aggregation, and if I add a terms aggregation to act on, I can't access the 2nd column value. That does bring up a point though. Even if I have a single column table with an aggregation, I would expect the context menu to display and allow me to send that single value. There is no way to do it right now.

Dosant commented:
@elastic-jb, 

> That does bring up a point though. Even if I have a single column table with an aggregation, I would expect the context menu to display and allow me to send that single value. There is no way to do it right now.

Correct, 
That is because for Drilldowns triggers we reused what already existed for filters. 
We didn't create a filter from an aggregated value from a table, so drill-downs inherited this limitation :( 
I think this has to be improved in scope of https://github.com/elastic/kibana/issues/78289 which we planned for 7.11 work. 


Regarding the current pr and helpers: this future table-vis work doesn't block helpers.

streamich commented on x-pack/plugins/ui_actions_enhanced/public/drilldowns/url_drilldown/url_template.ts:
```diff
@@ -69,6 +70,63 @@ handlebars.registerHelper('date', (...args) => {
+  return Array.from(valueString.matchAll(regexp));
+});
+
+function toString(value: unknown): string {
+  return String(value);
+}
+handlebars.registerHelper('lowercase', (rawValue: unknown) => toString(rawValue).toLowerCase());
```
Curious, does wrapping `String()` into `toString()` change something here?

```suggestion
handlebars.registerHelper('lowercase', (rawValue: unknown) => String(rawValue).toLowerCase());
```

streamich commented on x-pack/plugins/ui_actions_enhanced/public/drilldowns/url_drilldown/url_template.ts:
```diff
@@ -69,6 +70,63 @@ handlebars.registerHelper('date', (...args) => {
+handlebars.registerHelper('right', (rawValue: unknown, numberOfChars: number) => {
+  if (typeof numberOfChars !== 'number')
+    throw new Error('[left]: expected "number of characters to extract" to be a number');
+  return toString(rawValue).slice(-1 * numberOfChars);
```
```suggestion
  return toString(rawValue).slice(-numberOfChars);
```

streamich reviewed (approved):
LGTM.

Dosant commented:
I removed `matched` because using regex as part of an API was a bad idea in a first place. If anyone would really want to have something like this this would be possible with extensions: https://github.com/elastic/kibana/issues/78404

Instead of `match` I added `replace` which allows to replace or remove a substring.

elastic-jb commented:
Anton, the example I was thinking of was something like if I have a field that has "47.766201,-122.257057" stored as a single string field, it would be useful if I could have something that's kind of a cross between split and event.row so I could express that I want to use the value before the comma as a parameter, and the value after the comma as a separate parameter to send them to an API that requires them separately.

kibanamachine commented:
## :yellow_heart: Build succeeded, but was flaky
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/82184/)
* Commit: 4213b66ff971e40c200fa18b4f5cd34381fedac1
* Flaky suites:
  * xpack-kibana-ciGroup8

---

### [Test Failures](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/82184/testReport)

<details><summary>Chrome X-Pack UI Functional Tests.x-pack/test/functional/apps/discover/async_scripted_fields·js.discover async search with scripted fields query return results with valid scripted field</summary>

[Link to Jenkins](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/82184/testReport//junit/Chrome%20X-Pack%20UI%20Functional%20Tests/x-pack_test_functional_apps_discover_async_scripted_fields·js/discover_async_search_with_scripted_fields_query_return_results_with_valid_scripted_field)


#### Standard Out
```
Failed Tests Reporter:
  - Test has failed 4 times on tracked branches: https://github.com/elastic/kibana/issues/78553

[00:00:00]       │
[00:00:00]         └-: discover
[00:00:00]           └-> "before all" hook
[00:08:07]           └-: async search with scripted fields
[00:08:07]             └-> "before all" hook
[00:08:07]             └-> "before all" hook
[00:08:07]               │ info [kibana_scripted_fields_on_logstash] Loading "mappings.json"
[00:08:07]               │ info [kibana_scripted_fields_on_logstash] Loading "data.json.gz"
[00:08:07]               │ info [o.e.c.m.MetadataDeleteIndexService] [kibana-ci-immutable-ubuntu-16-tests-xxl-1602860447535957388] [.kibana_2/FZneRks_RFaD2Lp3LOLxRw] deleting index
[00:08:07]               │ info [o.e.c.m.MetadataDeleteIndexService] [kibana-ci-immutable-ubuntu-16-tests-xxl-1602860447535957388] [.kibana_1/LI7G96lqTl-sl2IQnvdU1w] deleting index
[00:08:07]               │ info [kibana_scripted_fields_on_logstash] Deleted existing index [".kibana_2",".kibana_1"]
[00:08:07]               │ info [o.e.c.m.MetadataCreateIndexService] [kibana-ci-immutable-ubuntu-16-tests-xxl-1602860447535957388] [.kibana_1] creating index, cause [api], templates [], shards [1]/[0]
[00:08:07]               │ info [o.e.c.r.a.AllocationService] [kibana-ci-immutable-ubuntu-16-tests-xxl-1602860447535957388] current.health="GREEN" message="Cluster health status changed from [YELLOW] to [GREEN] (reason: [shards started [[.kibana_1][0]]])." previous.health="YELLOW" reason="shards started [[.kibana_1][0]]"
[00:08:07]               │ info [kibana_scripted_fields_on_logstash] Created index ".kibana_1"
[00:08:07]               │ debg [kibana_scripted_fields_on_logstash] ".kibana_1" settings {"index":{"auto_expand_replicas":"0-1","number_of_replicas":"0","number_of_shards":"1"}}
[00:08:07]               │ info [kibana_scripted_fields_on_logstash] Indexed 19 docs into ".kibana_1"
[00:08:07]               │ info [o.e.c.m.MetadataMappingService] [kibana-ci-immutable-ubuntu-16-tests-xxl-1602860447535957388] [.kibana_1/8kvTDb1iTayNc9tM3oFJ0Q] update_mapping [_doc]
[00:08:07]               │ debg Migrating saved objects
[00:08:07]               │ proc [kibana]   log   [15:41:42.333] [info][savedobjects-service] Creating index .kibana_2.
[00:08:07]               │ info [o.e.c.m.MetadataCreateIndexService] [kibana-ci-immutable-ubuntu-16-tests-xxl-1602860447535957388] [.kibana_2] creating index, cause [api], templates [], shards [1]/[1]
[00:08:07]               │ info [o.e.c.r.a.AllocationService] [kibana-ci-immutable-ubuntu-16-tests-xxl-1602860447535957388] updating number_of_replicas to [0] for indices [.kibana_2]
[00:08:08]               │ info [o.e.c.r.a.AllocationService] [kibana-ci-immutable-ubuntu-16-tests-xxl-1602860447535957388] current.health="GREEN" message="Cluster health status changed from [YELLOW] to [GREEN] (reason: [shards started [[.kibana_2][0]]])." previous.health="YELLOW" reason="shards started [[.kibana_2][0]]"
[00:08:08]               │ proc [kibana]   log   [15:41:42.406] [info][savedobjects-service] Migrating .kibana_1 saved objects to .kibana_2
[00:08:08]               │ info [o.e.c.m.MetadataMappingService] [kibana-ci-immutable-ubuntu-16-tests-xxl-1602860447535957388] [.kibana_2/XHFDFTfjQq2yGIlaDylQvQ] update_mapping [_doc]
[00:08:08]               │ info [o.e.c.m.MetadataMappingService] [kibana-ci-immutable-ubuntu-16-tests-xxl-1602860447535957388] [.kibana_2/XHFDFTfjQq2yGIlaDylQvQ] update_mapping [_doc]
[00:08:08]               │ info [o.e.c.m.MetadataMappingService] [kibana-ci-immutable-ubuntu-16-tests-xxl-1602860447535957388] [.kibana_2/XHFDFTfjQq2yGIlaDylQvQ] update_mapping [_doc]
[00:08:08]               │ proc [kibana]   log   [15:41:42.603] [info][savedobjects-service] Pointing alias .kibana to .kibana_2.
[00:08:08]               │ proc [kibana]   log   [15:41:42.648] [info][savedobjects-service] Finished in 316ms.
[00:08:08]               │ debg applying update to kibana config: {"accessibility:disableAnimations":true,"dateFormat:tz":"UTC"}
[00:08:10]               │ info [logstash_functional] Loading "mappings.json"
[00:08:10]               │ info [logstash_functional] Loading "data.json.gz"
[00:08:10]               │ info [logstash_functional] Skipped restore for existing index "logstash-2015.09.22"
[00:08:10]               │ info [logstash_functional] Skipped restore for existing index "logstash-2015.09.20"
[00:08:10]               │ info [logstash_functional] Skipped restore for existing index "logstash-2015.09.21"
[00:08:11]               │ debg set roles = test_logstash_reader,global_discover_read
[00:08:11]               │ debg creating user test_user
[00:08:11]               │ info [o.e.x.s.a.u.TransportPutUserAction] [kibana-ci-immutable-ubuntu-16-tests-xxl-1602860447535957388] updated user [test_user]
[00:08:11]               │ debg created user test_user
[00:08:11]               │ debg TestSubjects.exists(kibanaChrome)
[00:08:11]               │ debg Find.existsByCssSelector('[data-test-subj="kibanaChrome"]') with timeout=2500
[00:08:11]               │ debg TestSubjects.find(kibanaChrome)
[00:08:11]               │ debg Find.findByCssSelector('[data-test-subj="kibanaChrome"]') with timeout=100000
[00:08:12]               │ debg applying update to kibana config: {"timepicker:timeDefaults":"{  \"from\": \"Sep 18, 2015 @ 19:37:13.000\",  \"to\": \"Sep 23, 2015 @ 02:30:09.000\"}"}
[00:08:12]               │ debg browser[INFO] http://localhost:6161/spaces/space_selector#/view/0b07e4a0-0fc6-11eb-98af-e1bdc0f99113?_g=(filters:!(),refreshInterval:(pause:!t,value:0),time:(from:now-15m,to:now))&_a=(columns:!(_source),filters:!(),index:'logstash-*',interval:auto,query:(language:kuery,query:''),sort:!()) 341 Refused to execute inline script because it violates the following Content Security Policy directive: "script-src 'unsafe-eval' 'self'". Either the 'unsafe-inline' keyword, a hash ('sha256-P5polb1UreUSOe5V/Pv7tc+yeZuJXiOi/3fqhGsU7BE='), or a nonce ('nonce-...') is required to enable inline execution.
[00:08:12]               │
[00:08:12]               │ debg browser[INFO] http://localhost:6161/bootstrap.js 42:19 "^ A single error about an inline script not firing due to content security policy is expected!"
[00:08:13]             └-> query should show failed shards pop up
[00:08:13]               └-> "before each" hook: global before each
[00:08:13]               │ debg navigating to discover url: http://localhost:6161/app/discover#/
[00:08:13]               │ debg navigate to: http://localhost:6161/app/discover#/
[00:08:13]               │ debg browser[INFO] http://localhost:6161/app/discover?_t=1602862907414#/ 341 Refused to execute inline script because it violates the following Content Security Policy directive: "script-src 'unsafe-eval' 'self'". Either the 'unsafe-inline' keyword, a hash ('sha256-P5polb1UreUSOe5V/Pv7tc+yeZuJXiOi/3fqhGsU7BE='), or a nonce ('nonce-...') is required to enable inline execution.
[00:08:13]               │
[00:08:13]               │ debg browser[INFO] http://localhost:6161/bootstrap.js 42:19 "^ A single error about an inline script not firing due to content security policy is expected!"
[00:08:13]               │ debg ... sleep(700) start
[00:08:13]               │ debg ... sleep(700) end
[00:08:13]               │ debg returned from get, calling refresh
[00:08:14]               │ debg browser[INFO] http://localhost:6161/app/discover?_t=1602862907414#/ 341 Refused to execute inline script because it violates the following Content Security Policy directive: "script-src 'unsafe-eval' 'self'". Either the 'unsafe-inline' keyword, a hash ('sha256-P5polb1UreUSOe5V/Pv7tc+yeZuJXiOi/3fqhGsU7BE='), or a nonce ('nonce-...') is required to enable inline execution.
[00:08:14]               │
[00:08:14]               │ debg browser[INFO] http://localhost:6161/bootstrap.js 42:19 "^ A single error about an inline script not firing due to content security policy is expected!"
[00:08:14]               │ debg currentUrl = http://localhost:6161/app/discover#/
[00:08:14]               │          appUrl = http://localhost:6161/app/discover#/
[00:08:14]               │ debg TestSubjects.find(kibanaChrome)
[00:08:14]               │ debg Find.findByCssSelector('[data-test-subj="kibanaChrome"]') with timeout=60000
[00:08:15]               │ debg ... sleep(501) start
[00:08:15]               │ debg browser[WARNING] http://localhost:6161/37378/bundles/kbn-ui-shared-deps/kbn-ui-shared-deps.js 20:69677 "Deprecation warning: value provided is not in a recognized RFC2822 or ISO format. moment construction falls back to js Date(), which is not reliable across all browsers and versions. Non RFC2822/ISO date formats are discouraged and will be removed in an upcoming major release. Please refer to http://momentjs.com/guides/#/warnings/js-date/ for more info.
[00:08:15]               │      Arguments: 
[00:08:15]               │      [0] _isAMomentObject: true, _isUTC: false, _useUTC: false, _l: undefined, _i: Sep 18, 2015 @ 19:37:13.000, _f: undefined, _strict: undefined, _locale: [object Object]
[00:08:15]               │      Error
[00:08:15]               │          at Function.createFromInputFallback (http://localhost:6161/37378/bundles/kbn-ui-shared-deps/kbn-ui-shared-deps.js:21:70106)
[00:08:15]               │          at http://localhost:6161/37378/bundles/kbn-ui-shared-deps/kbn-ui-shared-deps.js:21:90282
[00:08:15]               │          at http://localhost:6161/37378/bundles/kbn-ui-shared-deps/kbn-ui-shared-deps.js:21:90333
[00:08:15]               │          at Ct (http://localhost:6161/37378/bundles/kbn-ui-shared-deps/kbn-ui-shared-deps.js:21:90645)
[00:08:15]               │          at wt (http://localhost:6161/37378/bundles/kbn-ui-shared-deps/kbn-ui-shared-deps.js:21:90906)
[00:08:15]               │          at Mt (http://localhost:6161/37378/bundles/kbn-ui-shared-deps/kbn-ui-shared-deps.js:21:90988)
[00:08:15]               │          at r (http://localhost:6161/37378/bundles/kbn-ui-shared-deps/kbn-ui-shared-deps.js:21:67028)
[00:08:15]               │          at Object.parse (http://localhost:6161/37378/bundles/plugin/data/data.plugin.js:1:211420)
[00:08:15]               │          at validateTimeRange (http://localhost:6161/37378/bundles/plugin/data/data.plugin.js:1:173136)
[00:08:15]               │          at t._next (http://localhost:6161/37378/bundles/plugin/data/data.plugin.js:1:194578)"
[00:08:15]               │ debg ... sleep(501) end
[00:08:15]               │ debg in navigateTo url = http://localhost:6161/app/discover#/?_g=(filters:!(),refreshInterval:(pause:!t,value:0),time:(from:%27Sep%2018,%202015%20@%2019:37:13.000%27,to:%27Sep%2023,%202015%20@%2002:30:09.000%27))&_a=(columns:!(_source),filters:!(),index:eddf62c0-85bb-11ea-a555-0b6139dddc19,interval:auto,query:(language:kuery,query:%27%27),sort:!())
[00:08:15]               │ debg --- retry.try error: URL changed, waiting for it to settle
[00:08:16]               │ debg ... sleep(501) start
[00:08:16]               │ debg ... sleep(501) end
[00:08:16]               │ debg in navigateTo url = http://localhost:6161/app/discover#/?_g=(filters:!(),refreshInterval:(pause:!t,value:0),time:(from:%27Sep%2018,%202015%20@%2019:37:13.000%27,to:%27Sep%2023,%202015%20@%2002:30:09.000%27))&_a=(columns:!(_source),filters:!(),index:eddf62c0-85bb-11ea-a555-0b6139dddc19,interval:auto,query:(language:kuery,query:%27%27),sort:!())
[00:08:16]               │ debg TestSubjects.exists(statusPageContainer)
[00:08:16]               │ debg Find.existsByDisplayedByCssSelector('[data-test-subj="statusPageContainer"]') with timeout=2500
[00:08:19]               │ debg --- retry.tryForTime error: [data-test-subj="statusPageContainer"] is not displayed
[00:08:19]               │ debg TestSubjects.click(indexPattern-switch-link)
[00:08:19]               │ debg Find.clickByCssSelector('[data-test-subj="indexPattern-switch-link"]') with timeout=10000
[00:08:19]               │ debg Find.findByCssSelector('[data-test-subj="indexPattern-switch-link"]') with timeout=10000
[00:08:20]               │ debg Find.setValue('[data-test-subj="indexPattern-switcher"] input', 'logsta*')
[00:08:20]               │ debg Find.findByCssSelector('[data-test-subj="indexPattern-switcher"] input') with timeout=10000
[00:08:20]               │ debg Find.clickByCssSelector('[data-test-subj="indexPattern-switcher"] [title="logsta*"]') with timeout=10000
[00:08:20]               │ debg Find.findByCssSelector('[data-test-subj="indexPattern-switcher"] [title="logsta*"]') with timeout=10000
[00:08:20]               │ debg isGlobalLoadingIndicatorVisible
[00:08:20]               │ debg TestSubjects.exists(globalLoadingIndicator)
[00:08:20]               │ debg Find.existsByDisplayedByCssSelector('[data-test-subj="globalLoadingIndicator"]') with timeout=1500
[00:08:20]               │ debg TestSubjects.exists(globalLoadingIndicator-hidden)
[00:08:20]               │ debg Find.existsByCssSelector('[data-test-subj="globalLoadingIndicator-hidden"]') with timeout=100000
[00:08:21]               │ debg TestSubjects.getVisibleText(euiToastHeader)
[00:08:21]               │ debg TestSubjects.find(euiToastHeader)
[00:08:21]               │ debg Find.findByCssSelector('[data-test-subj="euiToastHeader"]') with timeout=10000
[00:08:21]               │ debg 1 of 3 shards failed
[00:08:21]               └- ✓ pass  (8.1s) "discover async search with scripted fields query should show failed shards pop up"
[00:08:21]             └-> query return results with valid scripted field
[00:08:21]               └-> "before each" hook: global before each
[00:08:21]               │ debg TestSubjects.click(indexPattern-switch-link)
[00:08:21]               │ debg Find.clickByCssSelector('[data-test-subj="indexPattern-switch-link"]') with timeout=10000
[00:08:21]               │ debg Find.findByCssSelector('[data-test-subj="indexPattern-switch-link"]') with timeout=10000
[00:08:21]               │ debg Find.setValue('[data-test-subj="indexPattern-switcher"] input', 'logstash-*')
[00:08:21]               │ debg Find.findByCssSelector('[data-test-subj="indexPattern-switcher"] input') with timeout=10000
[00:08:21]               │ debg Find.clickByCssSelector('[data-test-subj="indexPattern-switcher"] [title="logstash-*"]') with timeout=10000
[00:08:21]               │ debg Find.findByCssSelector('[data-test-subj="indexPattern-switcher"] [title="logstash-*"]') with timeout=10000
[00:08:21]               │ debg isGlobalLoadingIndicatorVisible
[00:08:21]               │ debg TestSubjects.exists(globalLoadingIndicator)
[00:08:21]               │ debg Find.existsByDisplayedByCssSelector('[data-test-subj="globalLoadingIndicator"]') with timeout=1500
[00:08:21]               │ debg TestSubjects.exists(globalLoadingIndicator-hidden)
[00:08:21]               │ debg Find.existsByCssSelector('[data-test-subj="globalLoadingIndicator-hidden"]') with timeout=100000
[00:08:24]               │ debg QueryBar.setQuery(php* OR *jpg OR *css*)
[00:08:24]               │ debg TestSubjects.click(queryInput)
[00:08:24]               │ debg Find.clickByCssSelector('[data-test-subj="queryInput"]') with timeout=10000
[00:08:24]               │ debg Find.findByCssSelector('[data-test-subj="queryInput"]') with timeout=10000
[00:08:25]               │ debg TestSubjects.getAttribute(queryInput, value)
[00:08:25]               │ debg TestSubjects.find(queryInput)
[00:08:25]               │ debg Find.findByCssSelector('[data-test-subj="queryInput"]') with timeout=10000
[00:08:25]               │ debg TestSubjects.click(querySubmitButton)
[00:08:25]               │ debg Find.clickByCssSelector('[data-test-subj="querySubmitButton"]') with timeout=10000
[00:08:25]               │ debg Find.findByCssSelector('[data-test-subj="querySubmitButton"]') with timeout=10000
[00:08:25]               │ debg isGlobalLoadingIndicatorVisible
[00:08:25]               │ debg TestSubjects.exists(globalLoadingIndicator)
[00:08:25]               │ debg Find.existsByDisplayedByCssSelector('[data-test-subj="globalLoadingIndicator"]') with timeout=1500
[00:08:27]               │ debg TestSubjects.exists(globalLoadingIndicator-hidden)
[00:08:27]               │ debg Find.existsByCssSelector('[data-test-subj="globalLoadingIndicator-hidden"]') with timeout=100000
[00:08:28]               │ info [o.e.x.c.a.AsyncResultsService] [kibana-ci-immutable-ubuntu-16-tests-xxl-1602860447535957388] failed to update expiration time for async-search [FlFIenA4ZDVhVGhhV0hMQ1BiVnA3TGcceEtUcER5azJTXzZsbHl0Q2NJRk1CZzoxMTgxMQ==]
[00:08:28]               │      org.elasticsearch.transport.RemoteTransportException: [kibana-ci-immutable-ubuntu-16-tests-xxl-1602860447535957388][127.0.0.1:6163][indices:data/write/update[s]]
[00:08:28]               │      Caused by: org.elasticsearch.index.engine.VersionConflictEngineException: [QHzp8d5aThaWHLCPbVp7Lg]: version conflict, required seqNo [15], primary term [1]. current document has seqNo [16] and primary term [1]
[00:08:28]               │ proc [kibana]   log   [15:42:02.474] [error][data][elasticsearch] [resource_not_found_exception]: FlFIenA4ZDVhVGhhV0hMQ1BiVnA3TGcceEtUcER5azJTXzZsbHl0Q2NJRk1CZzoxMTgxMQ==
[00:08:28]               │ERROR browser[SEVERE] http://localhost:6161/internal/search/ese/FlFIenA4ZDVhVGhhV0hMQ1BiVnA3TGcceEtUcER5azJTXzZsbHl0Q2NJRk1CZzoxMTgxMQ== - Failed to load resource: the server responded with a status of 404 (Not Found)
[00:08:28]               │ debg TestSubjects.getVisibleText(discoverQueryHits)
[00:08:28]               │ debg TestSubjects.find(discoverQueryHits)
[00:08:28]               │ debg Find.findByCssSelector('[data-test-subj="discoverQueryHits"]') with timeout=10000
[00:08:38]               │ debg --- retry.tryForTime error: Waiting for element to be located By(css selector, [data-test-subj="discoverQueryHits"])
[00:08:38]               │      Wait timed out after 10056ms
[00:08:38]               │ debg isGlobalLoadingIndicatorVisible
[00:08:38]               │ debg TestSubjects.exists(globalLoadingIndicator)
[00:08:38]               │ debg Find.existsByDisplayedByCssSelector('[data-test-subj="globalLoadingIndicator"]') with timeout=1500
[00:08:40]               │ debg --- retry.tryForTime error: [data-test-subj="globalLoadingIndicator"] is not displayed
[00:08:40]               │ debg TestSubjects.exists(globalLoadingIndicator-hidden)
[00:08:40]               │ debg Find.existsByCssSelector('[data-test-subj="globalLoadingIndicator-hidden"]') with timeout=100000
[00:08:40]               │ debg TestSubjects.getVisibleText(discoverQueryHits)
[00:08:40]               │ debg TestSubjects.find(discoverQueryHits)
[00:08:40]               │ debg Find.findByCssSelector('[data-test-subj="discoverQueryHits"]') with timeout=10000
[00:08:50]               │ debg --- retry.tryForTime failed again with the same message...
[00:08:51]               │ debg isGlobalLoadingIndicatorVisible
[00:08:51]               │ debg TestSubjects.exists(globalLoadingIndicator)
[00:08:51]               │ debg Find.existsByDisplayedByCssSelector('[data-test-subj="globalLoadingIndicator"]') with timeout=1500
[00:08:53]               │ debg --- retry.tryForTime error: [data-test-subj="globalLoadingIndicator"] is not displayed
[00:08:53]               │ debg TestSubjects.exists(globalLoadingIndicator-hidden)
[00:08:53]               │ debg Find.existsByCssSelector('[data-test-subj="globalLoadingIndicator-hidden"]') with timeout=100000
[00:08:53]               │ debg TestSubjects.getVisibleText(discoverQueryHits)
[00:08:53]               │ debg TestSubjects.find(discoverQueryHits)
[00:08:53]               │ debg Find.findByCssSelector('[data-test-subj="discoverQueryHits"]') with timeout=10000
[00:09:03]               │ debg --- retry.tryForTime error: Waiting for element to be located By(css selector, [data-test-subj="discoverQueryHits"])
[00:09:03]               │      Wait timed out after 10040ms
[00:09:04]               │ info Taking screenshot "/dev/shm/workspace/parallel/6/kibana/x-pack/test/functional/screenshots/failure/discover async search with scripted fields query return results with valid scripted field.png"
[00:09:04]               │ info Current URL is: http://localhost:6161/app/discover#/?_g=(filters:!(),refreshInterval:(pause:!t,value:0),time:(from:%27Sep%2018,%202015%20@%2019:37:13.000%27,to:%27Sep%2023,%202015%20@%2002:30:09.000%27))&_a=(columns:!(_source),filters:!(),index:eddf62c0-85bb-11ea-a555-0b6139dddc19,interval:auto,query:(language:kuery,query:%27php*%20OR%20*jpg%20OR%20*css*%27),sort:!())
[00:09:04]               │ info Saving page source to: /dev/shm/workspace/parallel/6/kibana/x-pack/test/functional/failure_debug/html/discover async search with scripted fields query return results with valid scripted field.html
[00:09:04]               └- ✖ fail: discover async search with scripted fields query return results with valid scripted field
[00:09:04]               │      Error: retry.tryForTime timeout: TimeoutError: Waiting for element to be located By(css selector, [data-test-subj="discoverQueryHits"])
[00:09:04]               │ Wait timed out after 10040ms
[00:09:04]               │     at /dev/shm/workspace/kibana/node_modules/selenium-webdriver/lib/webdriver.js:842:17
[00:09:04]               │     at process._tickCallback (internal/process/next_tick.js:68:7)
[00:09:04]               │       at onFailure (/dev/shm/workspace/parallel/6/kibana/test/common/services/retry/retry_for_success.ts:28:9)
[00:09:04]               │       at retryForSuccess (/dev/shm/workspace/parallel/6/kibana/test/common/services/retry/retry_for_success.ts:68:13)
[00:09:04]               │ 
[00:09:04]               │ 

```

#### Stack Trace
```
Error: retry.tryForTime timeout: TimeoutError: Waiting for element to be located By(css selector, [data-test-subj="discoverQueryHits"])
Wait timed out after 10040ms
    at /dev/shm/workspace/kibana/node_modules/selenium-webdriver/lib/webdriver.js:842:17
    at process._tickCallback (internal/process/next_tick.js:68:7)
    at onFailure (/dev/shm/workspace/parallel/6/kibana/test/common/services/retry/retry_for_success.ts:28:9)
    at retryForSuccess (/dev/shm/workspace/parallel/6/kibana/test/common/services/retry/retry_for_success.ts:68:13)
```
</details>

---

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### page load bundle size
| id | [before](https://github.com/elastic/kibana/commit/949c5a55b2d49768c744476cd35d23bea70a7a77) | [after](https://github.com/elastic/kibana/commit/4213b66ff971e40c200fa18b4f5cd34381fedac1) | diff |
| --- | --- | --- | --- |
| `uiActionsEnhanced` | 327.3KB | 330.0KB | +2.6KB |

### History
* :green_heart: [Build #81585](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/81585/) succeeded 10c50172d90ad5daa6f7adbba24087dfd8501228

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/81585/","number":"81585","commit":"10c50172d90ad5daa6f7adbba24087dfd8501228"},{"status":"UNSTABLE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/82184/","number":"82184","commit":"4213b66ff971e40c200fa18b4f5cd34381fedac1"}]}
PIPELINE-->

elastic-jb reviewed (approved):
Dropping match and adding replace makes sense. I think we have a good core here.

Dosant commented:
@elasticmachine merge upstream

Dosant commented:
> Anton, the example I was thinking of was something like if I have a field that has "47.766201,-122.257057" stored as a single string field, it would be useful if I could have something that's kind of a cross between split and event.row so I could express that I want to use the value before the comma as a parameter, and the value after the comma as a separate parameter to send them to an API that requires them separately.

@elastic-jb, I added `split` 👍

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/82373/)
* Commit: c26311711e584ac97af8e48e737361b6d5717af1

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### page load bundle size
| id | [before](https://github.com/elastic/kibana/commit/5858dd8f98153ea46f6350a5b04799541de65e1d) | [after](https://github.com/elastic/kibana/commit/c26311711e584ac97af8e48e737361b6d5717af1) | diff |
| --- | --- | --- | --- |
| `uiActionsEnhanced` | 327.3KB | 330.0KB | +2.6KB |

### History
* :yellow_heart: [Build #82184](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/82184/) was flaky 4213b66ff971e40c200fa18b4f5cd34381fedac1
* :green_heart: [Build #81585](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/81585/) succeeded 10c50172d90ad5daa6f7adbba24087dfd8501228

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/81585/","number":"81585","commit":"10c50172d90ad5daa6f7adbba24087dfd8501228"},{"status":"UNSTABLE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/82184/","number":"82184","commit":"4213b66ff971e40c200fa18b4f5cd34381fedac1"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/82373/","number":"82373","commit":"c26311711e584ac97af8e48e737361b6d5717af1"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/82391/)
* Commit: 2b14a9ff2eaaab041e964e4efe329e98dc89f6e8

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### page load bundle size
| id | [before](https://github.com/elastic/kibana/commit/5858dd8f98153ea46f6350a5b04799541de65e1d) | [after](https://github.com/elastic/kibana/commit/2b14a9ff2eaaab041e964e4efe329e98dc89f6e8) | diff |
| --- | --- | --- | --- |
| `uiActionsEnhanced` | 327.3KB | 330.4KB | +3.0KB |

### History
* :green_heart: [Build #82373](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/82373/) succeeded c26311711e584ac97af8e48e737361b6d5717af1
* :yellow_heart: [Build #82184](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/82184/) was flaky 4213b66ff971e40c200fa18b4f5cd34381fedac1
* :green_heart: [Build #81585](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/81585/) succeeded 10c50172d90ad5daa6f7adbba24087dfd8501228

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/81585/","number":"81585","commit":"10c50172d90ad5daa6f7adbba24087dfd8501228"},{"status":"UNSTABLE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/82184/","number":"82184","commit":"4213b66ff971e40c200fa18b4f5cd34381fedac1"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/82373/","number":"82373","commit":"c26311711e584ac97af8e48e737361b6d5717af1"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/82391/","number":"82391","commit":"2b14a9ff2eaaab041e964e4efe329e98dc89f6e8"}]}
PIPELINE-->

Dosant commented:
@elasticmachine merge upstream
